### PR TITLE
collectives: clarify src buffer entry requirements

### PR DESCRIPTION
**Pull request by davidozog**
# Summary of changes
This attempts to clarify the source buffer requirements upon entry to collectives and aligns the text across all relevant collective routines.  The discussion arose from https://github.com/openshmem-org/specification/pull/488

# Proposal Checklist
- [x] Link to issue(s)
- [ ] Changelog entry
- [ ] Reviewed for changes to front matter
- [ ] Reviewed for changes to back matter

### CODE REVIEW

File: content/shmem_alltoall.tex [hunk 1]
```diff
@@ -100,12 +100,16 @@
...
-    the following conditions must be ensured:
+    Before any \ac{PE} calls a \FUNC{shmem\_alltoall} routine, the following
+    conditions must be ensured, otherwise the behavior is undefined:
     \begin{itemize}
-    	\item The \VAR{dest} data object on all \acp{PE} in the team is
-    	ready to accept the \FUNC{shmem\_alltoall} data.
+        \item The \dest{} array on all \acp{PE} in the team is ready to
+            accept the result of the operation.
+        \item The \source{} buffer at the local \ac{PE} is ready to be
+            read by any \ac{PE} in the team.
```
Comment (maawad) [inline content/shmem_alltoall.tex:109]: Is there a particular reason for interchanging "buffer" and "array"? (dest array vs. source buffer)?

In the arguments description both are "Symmetric address of an array"

**Review by maawad — APPROVED** [review]
Looks good to me.

File: content/shmem_alltoall.tex (hunk 1 above)
Comment (davidozog) [reply content/shmem_alltoall.tex:109]: Good catch, I will change this to "array" throughout.

**Review by isubsmith — APPROVED** [review]

File: content/shmem_alltoall.tex (hunk 1 above)
Comment (davidozog) [reply content/shmem_alltoall.tex:109]: done

**Review by maawad — APPROVED** [review]

**Pull request merged**